core/deployment/src/main/java/io/quarkus/deployment/index/ApplicationArchiveBuildStep.java

```diff
@@ -173,7 +173,7 @@ static Path urlToPath(URL url) {
             } else if (url.getProtocol().equals("file")) {
                 int index = url.getPath().lastIndexOf("/META-INF");
                 if (index == -1) {
-                    return Paths.get(url.getPath());
+                    return Paths.get(url.toURI().getPath());
                 }
                 String pathString = url.getPath().substring(0, index);
```

Comment:
url.toURI()

Reply:
@machi1990 maybe let's create a variable for the `Path` once and for all? :)

Reply:
Thanks @Postremus for the review and confirmation.  I’ll revisit this and add more tests. 

@gsmet This is what I thought when looking at the codebase. I wanted to be sure if it fixes the issue above before. I’ll push my final changes later on.

Reply:
Done